row/mock/row_test.go

```diff
@@ -0,0 +1,337 @@
+		t.Error(s)
+	}
+}
+
+func TestAccess_NormalMocked_NormalUpdate(t *testing.T) {
+	p := Row{}
+	p.Mock(heptane.RowUpdate{Table: bogusTable, FieldValues: allFieldsValues}, errors.New("bogus"))
+	p.Mock(heptane.RowUpdate{Table: table, FieldValues: heptane.FieldValuesByName{}}, errors.New("bogus"))
+	p.Mock(heptane.RowUpdate{Table: table, FieldValues: allFieldsValues}, errors.New("err"))
+	a := heptane.RowUpdate{Table: table, FieldValues: allFieldsValues}
+	err := p.Access(a)
+	if err == nil {
+		t.Fatal(err)
+	}
+	if s := err.Error(); s != `err` {
+		t.Error(s)
+	}
+}
+
+func TestAccess_RefMocked_NormalUpdate(t *testing.T) {
+	p := Row{}
+	p.Mock(&heptane.RowUpdate{Table: bogusTable, FieldValues: allFieldsValues}, errors.New("bogus"))
+	p.Mock(&heptane.RowUpdate{Table: table, FieldValues: heptane.FieldValuesByName{}}, errors.New("bogus"))
+	p.Mock(&heptane.RowUpdate{Table: table, FieldValues: allFieldsValues}, errors.New("err"))
+	a := heptane.RowUpdate{Table: table, FieldValues: allFieldsValues}
+	err := p.Access(a)
+	if err == nil {
+		t.Fatal(err)
+	}
+	if s := err.Error(); s != `err` {
+		t.Error(s)
+	}
+}
+
+func TestAccess_NormalMocked_RefUpdate(t *testing.T) {
+	p := Row{}
+	p.Mock(heptane.RowUpdate{Table: bogusTable, FieldValues: allFieldsValues}, errors.New("bogus"))
+	p.Mock(heptane.RowUpdate{Table: table, FieldValues: heptane.FieldValuesByName{}}, errors.New("bogus"))
+	p.Mock(heptane.RowUpdate{Table: table, FieldValues: allFieldsValues}, errors.New("err"))
+	a := &heptane.RowUpdate{Table: table, FieldValues: allFieldsValues}
+	err := p.Access(a)
+	if err == nil {
+		t.Fatal(err)
+	}
+	if s := err.Error(); s != `err` {
+		t.Error(s)
+	}
+}
+
+func TestAccess_RefMocked_RefUpdate(t *testing.T) {
+	p := Row{}
+	p.Mock(&heptane.RowUpdate{Table: bogusTable, FieldValues: allFieldsValues}, errors.New("bogus"))
+	p.Mock(&heptane.RowUpdate{Table: table, FieldValues: heptane.FieldValuesByName{}}, errors.New("bogus"))
+	p.Mock(&heptane.RowUpdate{Table: table, FieldValues: allFieldsValues}, errors.New("err"))
+	a := &heptane.RowUpdate{Table: table, FieldValues: allFieldsValues}
+	err := p.Access(a)
+	if err == nil {
+		t.Fatal(err)
+	}
+	if s := err.Error(); s != `err` {
+		t.Error(s)
+	}
+}
+
+func TestAccess_Unmocked_RefDelete(t *testing.T) {
+	p := Row{}
+	a := &heptane.RowDelete{Table: table, FieldValues: allFieldsValues}
+	err := p.Access(a)
+	if err == nil {
+		t.Fatal(err)
+	}
+	if s := err.Error(); s != `Not Mocked: heptane.RowDelete{Table:heptane.Table{Name:"table", PartitionKey:[]heptane.FieldName{"foo"}, PrimaryKey:[]heptane.FieldName{"foo", "bar"}, Values:[]heptane.FieldName{"baz"}, Types:heptane.FieldTypesByName{"bar":"string", "baz":"int", "foo":"string"}, PrimaryKeyCachePrefix:[]heptane.CacheKey{"table_pk", "0"}}, FieldValues:heptane.FieldValuesByName{"bar":"2", "baz":3, "foo":"1"}}` {
```

Comment:
I see several comparisons like this. Should we compare literal long-strings like this? wouldn't it be more acceptable to define them as constants?

Reply:
This is a test file, two equal expected strings don't necessarily mean the same thing.